src/mono/wasm/runtime/lazyLoading.ts

```diff
@@ -21,7 +21,7 @@ export async function loadLazyAssembly(assemblyNameToLoad: string): Promise<bool
         behavior: "assembly",
     };
 
-    if (loaderHelpers.loadedAssemblies.some(f => f.includes(assemblyNameToLoad))) {
+    if (loaderHelpers.loadedAssemblies.includes(assemblyNameToLoad)) {
```

Comment:
I think original called with "Sample.App" matched  both "Sample.App.wasm" and "Sample.App.dll" for a reason.

Reply:
Discussed offline, that's not the case. Correct extension is required